Fix test-module failing to validate args

HD650 opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
The test-module pass a wrong argument _ansible_tmp cause the validation failed.
Change the argument _ansible_tmp to _ansible_tmpdir to fix this.
<!--- If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.-->
Fixes #40817 
Co-authored-by: Sviat <wk@sydorenko.org.ua>
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request


##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
test-module
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
devel
```


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful. -->
N/A
<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

Commit:
Fix test-module failing to validate args

The test-module pass a wrong argument _ansible_tmp cause the validation failed.
Change the argument _ansible_tmp to _ansible_tmpdir to fix this.

Commit:
Add a integration test for test-module.

Prior to this change, we don't have a test for test-module.

This change ensure the correctness of test-module script.

Commit:
debug commit

Commit:
debug commit
Prior to this change,

This change

abadger commented:
Tests are failing for an unrelated reason.

+1 to merge

sivel commented on test/integration/targets/test_infra/runme.sh:
```diff
@@ -21,3 +21,7 @@ echo "rc was $APB_RC (must be non-zero)"
+
+
+PING_MODULE_PATH="$(pwd)/../../../../lib/ansible/modules/system/ping.py"
+../../../../hacking/test-module -m "$PING_MODULE_PATH"
```
The tests are failing because freebsd doesn't have `/usr/bin/python`, so you will need to adjust this to use something like:

```
../../../../hacking/test-module -m "$PING_MODULE_PATH" -I ansible_python_interpreter=$(which python)
```

HD650 replied:
All right, I will try this, thanks!

Commit:
Fix the file doesn't access bug in freebsd

Prior to this change, test-module will raise "No such file" error since freebsd doesn't have /usr/bin/python.

Commit:
Improve code style

Make sure runme.sh can pass the shell sanity check.

webknjaz commented on test/integration/targets/test_infra/runme.sh:
```diff
@@ -21,3 +21,7 @@ echo "rc was $APB_RC (must be non-zero)"
+
+# ensure test-module script works well
+PING_MODULE_PATH="$(pwd)/../../../../lib/ansible/modules/system/ping.py"
+../../../../hacking/test-module -m "$PING_MODULE_PATH" -I ansible_python_interpreter=$(which python)
```
> test/integration/targets/test_infra/runme.sh:27:86: SC2046 Quote this to prevent word splitting.

@HD650 this linter message means that in the event of command in `$(...)` expression returning smth with space the command fill likely break. For example, if it returns `/usr/bin/env python` it will result in smth like `...  -I ansible_python_interpreter=/usr/bin/env python`, which is not what you want. Instead, you expect all of it to go to the same arg, like `...  -I ansible_python_interpreter="/usr/bin/env python"`.

To fix this, just wrap argument with double quotes `"` (single quotes prevent bash expression interpolation, which will break things).

abadger commented:
Good work!  Merged to devel for 2.7

webknjaz commented:
FTR: this regression has been introduced by https://github.com/ansible/ansible/pull/39833/files#diff-3934fe5332a78d0a3719f8e468f72272L128 18 days ago
